server/src/session.ts

```diff
@@ -340,29 +342,38 @@ export class Session {
     }
   }
 
-  private onDefinition(params: lsp.TextDocumentPositionParams) {
-    const {position, textDocument} = params;
-    const filePath = uriToFilePath(textDocument.uri);
-    const scriptInfo = this.projectService.getScriptInfo(filePath);
-    if (!scriptInfo) {
-      this.connection.console.log(`Script info not found for ${filePath}`);
+  private onDefinition(params: lsp.TextDocumentPositionParams): lsp.LocationLink[]|undefined {
+    const lsInfo = this.getLSPositionInfo(params);
+    if (lsInfo === undefined) {
       return;
     }
-
-    const {fileName} = scriptInfo;
-    const langSvc = this.getDefaultLanguageService(scriptInfo);
-    if (!langSvc) {
+    const {languageService, scriptInfo, offset} = lsInfo;
+    const definition = languageService.getDefinitionAndBoundSpan(scriptInfo.fileName, offset);
+    if (!definition || !definition.definitions) {
       return;
     }
+    const originSelectionRange = tsTextSpanToLspRange(scriptInfo, definition.textSpan);
+    const results: lsp.LocationLink[] = this.definitionsToLSPLocation(definition.definitions);
+    return results.map(r => ({...r, originSelectionRange}));
+  }
 
-    const offset = lspPositionToTsPosition(scriptInfo, position);
-    const definition = langSvc.getDefinitionAndBoundSpan(fileName, offset);
-    if (!definition || !definition.definitions) {
+  private onTypeDefinition(params: lsp.TextDocumentPositionParams): lsp.LocationLink[]|undefined {
+    const lsInfo = this.getLSPositionInfo(params);
+    if (lsInfo === undefined) {
       return;
     }
-    const originSelectionRange = tsTextSpanToLspRange(scriptInfo, definition.textSpan);
+    const {languageService, scriptInfo, offset} = lsInfo;
+    const definitions = languageService.getTypeDefinitionAtPosition(scriptInfo.fileName, offset);
+    if (!definitions) {
+      return;
+    }
+    return this.definitionsToLSPLocation(definitions);
+  }
+
+  private definitionsToLSPLocation(definitions: readonly ts.DefinitionInfo[]):
```

Comment:
1. Why not just return `lsp.LocationLink[]`?
2. Please consider renaming this to `tsDefinitionsToLspLocationLinks`

Reply:
1. I wanted to be more explicit about what was returned. Updated to accept optional `originSelectionRange`
2. done